File: app/src/main/java/com/rafaeltmbr/stopwatch/infra/presentation/mappers/impl/StringTimeMapperImpl.kt
package com.rafaeltmbr.stopwatch.infra.presentation.mappers.impl

import com.rafaeltmbr.stopwatch.infra.presentation.entities.ViewTime
import com.rafaeltmbr.stopwatch.infra.presentation.mappers.StringTimeMapper
import com.rafaeltmbr.stopwatch.infra.presentation.mappers.ViewTimeMapper

class StringTimeMapperImpl(
    private val viewTimerMapper: ViewTimeMapper
) : StringTimeMapper {
    override fun mapToStringTime(milliseconds: Long): String {
        val viewTime: ViewTime = viewTimerMapper.mapToViewTime(milliseconds)
        return "%s:%s.%s".format(
            viewTime.minutes.joinToString(""),
            viewTime.seconds.joinToString(""),
            viewTime.fraction.joinToString("")
        )
    }
}

File: ios/Stopwatch/platform/presentation/mappers/impl/StringTimeMapperImpl.swift
class StringTimeMapperImpl: StringTimeMapper {
    var viewTimeMapper: ViewTimeMapper
    
    init(_ viewTimeMapper: ViewTimeMapper) {
        self.viewTimeMapper = viewTimeMapper
    }
    
    func map(_ milliseconds: Int) -> String {
        let viewTime = viewTimeMapper.map(milliseconds)
        let minutes = viewTime.minutes.joined()
        let seconds = viewTime.seconds.joined()
        let fraction = viewTime.fraction.joined()
        return "\(minutes):\(seconds).\(fraction)"
    }
}
